### PR TITLE
Speech-to-Text automatic punctuation, model selection

### PR DESCRIPTION
**Pull request by telpirion**
This adds the following samples:

| Region tag | Python canonical |
| ----------- | ----------------- |
| speech_transcribe_model_selection | [sample](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/speech/cloud-client/transcribe_model_selection.py#L30-L55) |
| speech_transcribe_auto_punctuation | [sample](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/speech/cloud-client/beta_snippets.py#L111-L137) |

### CODE REVIEW

**Comment by telpirion** [comment]
You know what: the enhanced model test won't pass until we enable data logging on the test project.

https://cloud.google.com/speech-to-text/docs/enable-data-logging

**Comment by beccasaurus** [comment]
~ _Happy Monday_ ~ `#ping`

File: speech/api/Recognize/Recognize.cs [hunk 1]
```diff
@@ -170,6 +176,54 @@ static object SyncRecognizeWords(string filePath)
...
         // [END speech_sync_recognize_words]
 
+        // [START speech_transcribe_auto_punctuation]
+        static object SyncRecognizePunctuation(string filePath)
+        {
+            var speech = SpeechClient.Create();
+            var response = speech.Recognize(new RecognitionConfig()
+            {
+                Encoding = RecognitionConfig.Types.AudioEncoding.Linear16,
+                SampleRateHertz = 8000,
```
[the remark this answers is not in the document]
Comment (beccasaurus) [reply speech/api/Recognize/Recognize.cs]: Ah, nevermind, the canonical uses 8000 for these samples 🤷🏼‍♀️

**Comment by beccasaurus** [comment]
LGTM – mergeable once tests are passing

**Comment by SurferJeffAtGoogle** [comment]
Please update the README in the parent directory with instructions describing what needs to be enabled to run this sample.

**Comment by telpirion** [comment]
@beccasaurus : I swapped out the enhanced model sample for model selection. Please take a look at your earliest convenience.

Thank you very much!

**Pull request merged**